src/components/Personalization/createModulesProvider.js

```diff
@@ -0,0 +1,21 @@
+import createPreprocessors from "./createPreprocessors";
+
+export default ({ modules, preprocessors = createPreprocessors() }) => {
+  const getModules = schema => {
+    return {
+      getAction: type => {
+        if (!modules[schema]) {
+          return undefined;
+        }
+
+        return modules[schema][type];
+      },
+      getPreprocessors: () => preprocessors[schema],
+      getSchema: () => schema
+    };
+  };
+
+  return {
+    getModules
+  };
+};
```

Comment:
Having a bunch of getters is a code smell. Consider just building an execute function that would run the preprocessors and the actions for an activity.